### PR TITLE
first week's update

### PR DESCRIPTION
**Pull request by sigma21**
Still has some glitches, I'll be updating.

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -30,18 +32,89 @@
...
+        repositories.forEach(repo => {
+          const options = createAndAppend('option', select, {
+            value: valueNumber,
+          });
+          options.textContent = repo.name;
+          valueNumber += 1;
+        });
+
+        const selectElement = document.getElementById('selectionTree');
+        selectElement.addEventListener('change', event => {
```
Comment (paulbremer) [inline homework/index.js]: On every change event you are appending the info about the repoContributors but not removing the info about the previous selected repo so the contributors get added to a long list, not from the repo selected. Try to write some code to remove the repoContributors from the previous one and only show the repoContributors from the selected repo!

**Comment by wouterkleijn** [comment]
Thanks @paulbremer For all the feedback! @sigma21 Would be great if you could check the feedback, respond to it and implement changes!

File: homework/index.js [hunk 2]
```diff
@@ -30,18 +32,89 @@
...
+        const repositories = JSON.parse(jsonString, null);
+
+        let contributorsURL = '';
+
+        repositories.forEach(repo => {
+          const options = createAndAppend('option', select, {
+            value: valueNumber,
+          });
+          options.textContent = repo.name;
+          valueNumber += 1;
```
[the remark this answers is not in the document]
Comment (sigma21) [reply homework/index.js]: Changed my code according to your advice. Thanks!

File: homework/index.js [hunk 3]
```diff
@@ -30,18 +32,89 @@
...
+        });
+        let valueNumber = 0;
+
+        const repositories = JSON.parse(jsonString, null);
+
+        let contributorsURL = '';
+
+        repositories.forEach(repo => {
+          const options = createAndAppend('option', select, {
+            value: valueNumber,
```
[the remark this answers is not in the document]
Comment (sigma21) [reply homework/index.js]: I rewrote it your way. Thanks!

File: homework/index.js [hunk 4]
```diff
@@ -30,18 +32,89 @@
...
+              const json = JSON.stringify(data1, null, 2);
+              const repoDetail = JSON.parse(json, null);
+
+              const repoDetails = document.getElementById('repoContributors');
+              createAndAppend('h2', repoDetails, {
+                text: 'Contributors',
+              });
+              repoDetail.forEach(repo => {
+                createAndAppend('img', repoDetails, {
+                  src: `${repo.avatar_url}`,
```
[the remark this answers is not in the document]
Comment (sigma21) [reply homework/index.js]: Right, no need for quotes. Fixed it. Thanks!

File: homework/index.js [hunk 5]
```diff
@@ -30,18 +32,89 @@
...
+                  style: 'width:100px;',
```
[the remark this answers is not in the document]
Comment (sigma21) [reply homework/index.js]: Moved it to CSS :)

File: homework/style.css [hunk 6]
```diff
@@ -1,3 +1,20 @@
 .alert-error {
-    color: red;
-}
\ No newline at end of file
+  color: red;
+}
+
+#root {
+  display: flex;
```
[the remark this answers is not in the document]
Comment (sigma21) [reply homework/style.css]: I've added responsiveness in week2 update.

File: homework/index.js (hunk 1 above)
Comment (sigma21) [reply homework/index.js]: I've fixed this issue on week2 update. Added a function that deletes child element before adding the new contributor div.

File: homework/index.js [hunk 7]
```diff
@@ -30,18 +32,89 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
+  function main() {
```
[the remark this answers is not in the document]
Comment (sigma21) [reply homework/index.js]: Thanks I'll try to do it that way in this week's update.

**Comment by sigma21** [comment]
Hey @paulbremer thanks for the feedback! I've made the changes requested. Hopefully it'll look better in this week's update.

**Comment by paulbremer** [comment]
Hey @sigma21 looking forward to your next homework ;)! PR approved!

**Pull request closed without merging**